src/proto/beats.proto

```diff
@@ -10,22 +11,75 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum SectionType {
+  INTRO = 0;
+  OUTRO = 1;
+  OTHER = 2;
+}
+
+// Setting grey as the default color
+message Color {
+  optional int32 red = 1 [ default = 100 ];
+  optional int32 green = 2 [ default = 100 ];
+  optional int32 blue = 3 [ default = 100 ];
 }
 
 message Bpm {
-  optional double bpm = 1;
-  optional Source source = 2 [ default = ANALYZER ];
+  optional double bpm = 1 [ default = 0 ];
+  optional Source source = 2;
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message BpmMarker {
+  optional int32 beat_index = 1;
+  optional Bpm bpm = 2;
 }
 
-message BeatGrid {
-  optional Bpm bpm = 1;
-  optional Beat first_beat = 2;
+message TimeSignature {
+  optional int32 beats_per_bar = 1 [ default = 4];
+  optional int32 note_value = 2 [ default = 4 ];
 }
```

Comment:
Here we need something to annotate the accented beats we want to see on the waveforms. 
This is required to annotate odd time signatures with high denominators. Like the Igorrr track.  

I like the idea of @crisclacerda  to use the Shorthand Notation for Musical Rhythm (SNMR)
Either in the ASCII or binary format. 
Igorrr:  
14/16
HH--
10001000100100

Reply:
We have no consensus that such complexity is wanted.

Reply:
Are you still the opinion to annotate the Igorrr track with changing speeds inside a measure?

Reply:
I never suggested that.

Reply:
What did you suggest?

Reply:
I believe we could annotate it in 3 ways, odd signature, changing tempo inside a measure or the note simply being outside the beat. These all in way makes sense, and unless one have to the original sheet music, or the daw project there is no possible way of knowing which one is the "ground truth". What is important for us are the use cases, sync and quantizing.

The odd signature allows to quantize to the notes without add messing to sync.
The changing tempo inside the measure also allows us to quantize nicely but will cause sync to warp the beats.
Finally the note being outside the beat does not allows to sync and does not allow to quantize either.

Ideally we should be flexible to support all. 

For now to support the odd signatures all we need is an extra 32 bit int together with the time signature in the proto to save the SNMR binary format. Then we need some code to hide some beats based on that, and some way for the user to add or edit those.

Reply:
As it is no difference between a int32 and a int 64 inside the prototype let's go with int64 this way we have without a doubt enough beats.

Reply:
>  The changing tempo inside the measure also allows us to quantize nicely but will cause sync to warp the beats.

Right, marking a complex rhythm as changing tempos does not make sense. I never suggested this.

> For now to support the odd signatures all we need is an extra 32 bit int together with the time signature in the proto to save the SNMR binary format. Then we need some code to hide some beats based on that, and some way for the user to add or edit those.

I don't agree that this is necessary nor do I understand why it is desirable. I would appreciate if we drop this idea at least for now and focus on getting this done. More features for edge cases can be added later as long as we don't encode assumptions into the data structure that prevent them.

Reply:
> Right, marking a complex rhythm as changing tempos does not make sense. I never suggested this.

Cool, we have a common sense here. But how do you like to annotate the Igorrr track? 

> I don't agree that this is necessary nor do I understand why it is desirable.

Did you read my Zulip post whit the screen shots of down beat-matched tracks? Down-beat matching is the mayor use case for me. I am happy to discuss alternative Ideas, but for now I don't see any better way. 

> More features for edge cases can be added later as long as we don't encode assumptions into the data structure that prevent them.

Unfortunately this is not an edge case. The way how and if we represent different rhythms in a measure is essential for the workflow. Sure, the Igorrr track is an edge case with no real value for beat matching, but if we solve this problem everything else is solved. Real use cases are 9/8 and 7/8 tracks which should receive first class support among easy 4/4 and 3/4 tracks.

Reply:
```
optional int64 snmr_bins = 3 [ default = 0 ]
```
Binary Shorthand Notation for Musical Rhythm in binary format.

0 = invalid value, all metronome beats annotated.